Documentation/CrossPlatform/LeapMotionMRTK.md

```diff
@@ -43,52 +50,88 @@ The Leap Motion Data Provider enables articulated hand tracking for VR and could
 
+    
 1. Testing the Leap Motion Data Provider
     - After Leap Motion Data Provider has been added to the input system profile, press play, move your hand in front of the Leap Motion Controller and you should see the joint representation of the hand.
 
 1. Building your project 
     - Navigate to **File > Build Settings**
     - Only Standalone builds are supported if using the Leap Motion Data Provider.
+    - For instructions on how to use a Windows Mixed Reality headset for Standalone builds, see [Build and Deploy](../BuildAndDeploy#building-and-deploying-mrtk-to-a-windows-mixed-reality-headset).
 
 ## Getting the hand joints 
 
 Getting joints using the Leap Motion Data Provider is identical to hand joint retrieval for an MRTK Articulated Hand.  For more information, see [Hand Tracking](../Input/HandTracking.md#polling-joint-pose-from-handjointutils).
 
-The following is a simple example of how to retrieve the pose of the palm joint and adds a sphere to follow the hand joint.
+With MRTK in a unity scene and the Leap Motion Data Provider added as an Input Data Provider in the Input System profile, create an empty game object and attach the following example script.
+
+This script is a simple example of how to retrieve the pose of the palm joint in a Leap Motion Hand.  A sphere follows the left Leap hand while a cube follows the right Leap hand.
+
 ```
+using Microsoft.MixedReality.Toolkit;
 using Microsoft.MixedReality.Toolkit.Input;
 using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections.Generic;
 using UnityEngine;
 
-public class HandJointsLeap : MonoBehaviour
+public class LeapHandJoints : MonoBehaviour, IMixedRealityHandJointHandler
 {
-    private GameObject sphere;
+    private GameObject leftHandSphere;
+    private GameObject rightHandCube;
 
     private void Start()
     {
-        sphere = GameObject.CreatePrimitive(PrimitiveType.Sphere);
-        sphere.transform.localScale = Vector3.one * 0.03f;
+        // Register the HandJointHandler as a global listener
+        CoreServices.InputSystem.RegisterHandler<IMixedRealityHandJointHandler>(this);
+
+        // Create a sphere to follow the left hand palm position
+        leftHandSphere = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+        leftHandSphere.transform.localScale = Vector3.one * 0.03f;
+
+        // Create a cube to follow the right hand palm position 
+        rightHandCube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+        rightHandCube.transform.localScale = Vector3.one * 0.03f;
     }
 
-    private void Update()
+    public void OnHandJointsUpdated(InputEventData<IDictionary<TrackedHandJoint, MixedRealityPose>> eventData)
     {
-        // If the hand joint pose is available set the position of the sphere to the palm of the right hand
-        if (HandJointUtils.TryGetJointPose(TrackedHandJoint.Palm, Handedness.Right, out MixedRealityPose index))
+        if (eventData.InputSource.SourceName == "Leap Left Controller")
```

Comment:
Do we want to specifically call out the string "Leap Left Controller" or should we look for handedness of the controller/event data instead?

In general it's very odd to see string based comparison for identity (i.e. "Leap Left Controller"), and if we have to do that, this shoudl be a constant exposed somewhere else in MRTK. People shouldn't be doing explicit string based compares to a locally defined string.

Reply:
Agreed, I think we generally shouldn't be recommending string comparisons. The event data should have handedness though! And then it becomes cross-compatible with any hand implementation that provides this data.

Reply:
Getting the leap hand by the controller name instead of handedness was done to intentionally exculde the input simulation hand.  

This is what happens when I use handedness instead of the controller name:

![leapInputSimulation](https://user-images.githubusercontent.com/53493796/80847181-81283180-8bc3-11ea-8caf-6ff518756294.gif)


But handedness can be used instead of the name

Reply:
Changed to handedness